sync.js

```diff
@@ -486,17 +486,21 @@ class Sync extends events.EventEmitter {
             const fileProjectId = (data && data.discoveryKey) || SYNC_DEFAULT_KEY
 
             if (ourProjectId !== fileProjectId) {
-              function formatId (id) {
-                return id === SYNC_DEFAULT_KEY ? id : id.slice(0, 7)
-              }
-
-              return onerror(
-                new Error(
-                  `trying to sync two different projects (us=${formatId(
-                    ourProjectId
-                  )}) (syncfile=${formatId(fileProjectId)})`
+              syncfile.close(() => {
+                onerror(
```

Comment:
felt like it made sense to place as a callback within `syncfile.close`, but theoretically seems like we could also just call it separately i.e.

```js
syncfile.close(() => {})

onerror(...)

return
```

not sure if that's considered idiomatic but just throwing it out there in case